akka-http-testkit/src/main/scala/akka/http/scaladsl/testkit/RouteTestResultComponent.scala

```diff
@@ -95,7 +97,11 @@ trait RouteTestResultComponent {
     private def failNeitherCompletedNorRejected(): Nothing =
       failTest("Request was neither completed nor rejected within " + timeout)
 
-    private def awaitAllElements[T](data: Source[T, _]): immutable.Seq[T] =
-      data.grouped(100000).runWith(Sink.head).awaitResult(timeout)
+    private def awaitAllElements[T](data: Source[T, _]): immutable.Seq[T] = {
+      data.grouped(100000).runWith(Sink.head).recover({
+        case e: NoSuchElementException ⇒ Nil
+      })(ExecutionContexts.sameThreadExecutionContext)
+        .awaitResult(timeout)
+    }
```

Comment:
Why not create a sink for this purpose instead? I don't like abusing exceptions like this.